src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2337,29 +2336,24 @@ protected int getCacheLimit() {
     })
     public abstract static class InvertAsciiCaseBytesNode extends RubyNode {
 
-        @CompilationFinal private boolean lowerToUpper;
-        @CompilationFinal private boolean upperToLower;
+        private final boolean lowerToUpper;
+        private final boolean upperToLower;
 
         public static InvertAsciiCaseBytesNode createLowerToUpper() {
-            final InvertAsciiCaseBytesNode ret = StringNodesFactory.InvertAsciiCaseBytesNodeGen.create(null, null);
-            ret.lowerToUpper = true;
-
-            return ret;
+            return StringNodesFactory.InvertAsciiCaseBytesNodeGen.create(true, false, null, null);
         }
 
         public static InvertAsciiCaseBytesNode createUpperToLower() {
-            final InvertAsciiCaseBytesNode ret = StringNodesFactory.InvertAsciiCaseBytesNodeGen.create(null, null);
-            ret.upperToLower = true;
-
-            return ret;
+            return StringNodesFactory.InvertAsciiCaseBytesNodeGen.create(false, true, null, null);
         }
 
         public static InvertAsciiCaseBytesNode createSwapCase() {
-            final InvertAsciiCaseBytesNode ret = StringNodesFactory.InvertAsciiCaseBytesNodeGen.create(null, null);
-            ret.lowerToUpper = true;
-            ret.upperToLower = true;
+            return StringNodesFactory.InvertAsciiCaseBytesNodeGen.create(true, true, null, null);
+        }
 
-            return ret;
+        protected InvertAsciiCaseBytesNode(boolean lowerToUpper, boolean upperToLower) {
+            this.lowerToUpper = lowerToUpper;
+            this.upperToLower = upperToLower;
         }
```

Comment:
Looks much nicer :)
InvertAsciiCaseNode could also benefit from a normal constructor.

Reply:
How so? It doesn't take any options.

Reply:
https://github.com/graalvm/truffleruby/pull/563/files#diff-c01f8a8b13ab8ed69470cc9c6e52e5b2R2407 could be just part of InvertAsciiCaseNode's constructor so a InvertAsciiCaseBytesNode can be passed to it.